src/main/java/Entities/Implementations/QuantityRecipeItem.java

```diff
@@ -0,0 +1,38 @@
+package Entities.Implementations;
+
+import Entities.Implementations.AbstractRecipeItem;
+import Entities.Ingredient;
+
+public class QuantityRecipeItem extends AbstractRecipeItem {
+    /**
+     * Implement a ItemImpl, giving the ingredient,
+     * quantity of this ingredient, and if the ingredient is optional.
+     *
+     * @param ingredient The ingredient
+     * @param amount     The quantity of this ingredient
+     * @param optional   Is the ingredient optional?
+     */
+    public QuantityRecipeItem(Ingredient ingredient, float amount, boolean optional) {
+        super(ingredient, amount, optional);
+
+    }
+
+    public String display() {
+        String stringquantity = Float.toString(this.quantity());
+        if (this.quantity() > 1) {
+
+            return stringquantity + this.ingredient.name() + "s";
+        }
+        if (this.quantity() < 1) {
+            //checks to see if the first letter of the ingredient name is a vowel.
+            char letter = (this.ingredient.name()).charAt(0);
+
+            if ("AEIOUaeiou".indexOf(letter) != -1) {
+                return stringquantity + "of an" + this.ingredient.name();
+            }
+            return stringquantity + "of a" + this.ingredient.name();
+        }
+        return stringquantity + this.ingredient.name();
```

Comment:
This method could be simplified.

Since the body of `<1` is the most  complex, it should not be within an if statement as this nests it further. Try checking `=1`, `>1` and then have the `<1` case not in an if or else statement.

You could also remove the return statement for `of an` and instead declare a string `of a`, if the first letter is a vowel append `n`, and then construct a string using this and return. Cuts out a return which makes it a bit easier to follow. See if it helps.